src/App.js

```diff
@@ -10,10 +10,138 @@ function App() {
     beginDate: '',
     endDate: ''
   });
-  const [glocationFilter, setGlocationFilter] = useState('');
-  const [newsDeskFilter, setNewsDeskFilter] = useState([]);
-  const [materialTypeFilter, setMaterialTypeFilter] = useState([]);
+  const [glocation, setGlocation] = useState('');
+  const [newsDesks, setNewsDesks] = useState([]);
+  const [materialTypes, setMaterialTypes] = useState([]);
   const [sortOrder, setSortOrder] = useState('relevance');
+  const [articles, setArticles] = useState(null);
+  const [totalHits, setTotalHits] = useState(0);
+  const [page, setPage] = useState(0);
+  const [isFetching, setIsFetching] = useState(false);
+  
+  const isMounted = useRef(false);
+
+  // If page changes, submit search
+  // Do nothing on initial render
+  useEffect(() => {
+    if (isMounted.current) {
+      fetchArticles();
+    }
+  }, [page]);
+  
+  // If `sortOrder` changes, either set page back to 0 or submit search
+  // Do nothing on initial render
+  useEffect(() => {
+    if (isMounted.current) {
+      if (page !== 0) {
+        setPage(0)
+      } else {
+        fetchArticles();
+      }
+    }
+  }, [sortOrder]);
+
+  // On initial render, set `isMounted` equal to true
+  useEffect(() => {
+    if (!isMounted.current) {
```

Comment:
man this is annoying, but not your fault

do you think the name current captures the meaning of this variable?